core/src/components/button/button.ionic.scss

```
@@ -53,45 +53,61 @@
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-solid.ion-color) {
+  --background-activated: #{globals.current-color(shade)};
+  --background-hover: #{globals.current-color(shade)};
+  --background: #{globals.current-color(base)};
+  --color: #{globals.current-color(contrast)};
+}
+
 // Outline Button
 // --------------------------------------------------
 
 :host(.button-outline) {
   --border-width: #{globals.$ion-border-size-025};
   --border-style: #{globals.$ion-border-style-solid};
-  --background-activated: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-activated: #{globals.$ion-bg-primary-subtle-press};
   --background-focused: transparent;
-  --background-hover: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-hover: #{globals.$ion-bg-primary-subtle-press};
   --background-hover-opacity: 1;
   --border-color: #{globals.ion-color(primary, base)};
   --color: #{globals.ion-color(primary, base)};
   --ripple-opacity: var(--background-activated-opacity, 1);
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-outline.ion-color) {
+  --background-activated: #{globals.current-color(shade, $subtle: true)};
+  --background-hover: #{globals.current-color(shade, $subtle: true)};
+  --border-color: #{globals.current-color(base)};
+  --color: #{globals.current-color(base)};
+}
+
+// Warning Outline Button - use foreground color for text and border
+:host(.button-outline.ion-color-warning) .button-native {
+  --ion-color-base: #{globals.ion-color(warning, foreground)};
```

We shouldn't be using `--ion-color-base` to override since that's not how we do it. We have border color and text color variables for this.


```suggestion
  --color: #{globals.ion-color(warning, foreground)};
  --border-color: #{globals.ion-color(warning, foreground)};
```